### DISCUSSION TITLE
scan: allow to skip xcodebuild -showBuildSettings command

### DISCUSSION THREAD

**Discussion by ghost** [fastlane/fastlane#20897] [Ideas]
Hi all 👋 
In multiple projects I have stumbled on issue regarding showBuildSettings in fastlane (especially fastlane scan).
It is quite common that command takes longer than anticipated ending up with unnecessary retries and sometimes quite a lot of wasted time.
I saw there were questions about that like [this issue](https://github.com/fastlane/fastlane/issues/14777) but it never ended in possibility to skip showBuildSetting entirely.

I looked into the codebase to see what are those settings needed for and it seems that for not so much.
It is used in two places in `detect_values.rb`:
- getting `derived_data_path` in none specified (`default_derived_data` func)
- getting `deployment_target_version` if none specified (`get_deployment_target_version` func)

It is indirectly used in `building_mac_catalyst_for_mac` func to check whether app supports mac catalyst as name implies.
It is also being invoked indirectly in core to get `app_name` if none specified (used just for Slack purposes).

Beside `building_mac_catalyst_for_mac` func all calls to `Scan.project.build_settings` can be omitted by providing value to `scan` command.
In case of `building_mac_catalyst_for_mac` build settings are not really needed because those are used in & bool expression (unfortunately in first place).

In order to allow for scan to run without asking for build settings which can be lengthy user could just provide all needed parameters.
I would like to merge small change allowing just for that :) 
In the future it may be worth to check other commands if showBuildSettings can be omitted there too.

**Comment by ghost** [comment]
Created PR for that: https://github.com/fastlane/fastlane/pull/20898

**Comment by irgaly** [comment]
FYI:

It seems that derived_data_path could be calculated from .xcodeproj path without `xcodebuld -showBuildSettings` invocation.

https://pewpewthespells.com/blog/xcode_deriveddata_hashes.html
